plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorExecutor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ui.editors.sql;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.model.struct.DBSObject;
+import org.jkiss.dbeaver.tasks.ui.nativetool.NativeToolWizardDialog;
+import org.jkiss.dbeaver.tasks.ui.wizard.TaskConfigurationWizard;
+import org.jkiss.dbeaver.ui.UIUtils;
+
+/**
+ * Allows opening native execution wizards
+ *
+ * @param <CONTAINER> container to read settings from
+ */
+public abstract class SQLEditorExecutor<CONTAINER extends DBSObject> {
+    /**
+     * Opens the wizard for the database
+     *
+     * @param container container to read settings from
+     * @param editor SQL editor
+     * @throws DBException if failed to open the wizard
+     */
+    public void execute(@NotNull CONTAINER container, @NotNull SQLEditor editor) throws DBException {
+        NativeToolWizardDialog dialog = new NativeToolWizardDialog(
+            UIUtils.getActiveWorkbenchWindow(),
+            createTaskConfigurationWizard(container, editor));
+        dialog.open();
+    }
+
+    @NotNull
+    protected abstract TaskConfigurationWizard<?> createTaskConfigurationWizard(@NotNull CONTAINER container,
+        @NotNull SQLEditor editor);
```

Comment:
Yes, no closing paren on a new line because it doesn't have a body.

```suggestion
    protected abstract TaskConfigurationWizard<?> createTaskConfigurationWizard(
        @NotNull CONTAINER container,
        @NotNull SQLEditor editor);
```